app/dashboard/src/utilities/LocalStorage.ts

```diff
@@ -74,9 +76,28 @@ export default class LocalStorage {
 
   /** Register runtime behavior associated with a {@link LocalStorageKey}. */
   static registerKey<K extends LocalStorageKey>(key: K, metadata: LocalStorageKeyMetadata<K>) {
+    if (!IS_DEV_MODE) {
+      invariant(
```

Comment:
this one looks... silly. I do want to know that before we deploy to prod :)

Reply:
ah, good point... i was hiding it because it was (incorrectly) showing up on HMR refresh. any thoughts? i could instead have it e.g. throw and error and catch it to capture the stack trace... it's hacky but *should* more or less work...

Reply:
Maybe we can utilize this API to avoid incorrect behavior on HMR? https://vitejs.dev/guide/api-hmr

Reply:
not sure it's an option, yeah. especially since (afaict) you'd have to pass in `import.meta.hot` from the module that is calling `.register`, which means exposing implementation details to implementors.

right now i'm pushing an implementation based on stack traces instead, let me know your thoughts i guess?

Reply:
In dev mode, I'm okay with that.

FWIW, to get access to the stack trace, you can call `new Error().stack` instead

Reply:
yeah XD i was throwing a new error and then wondering why i was catching it when i could just access its stack directly